Source/WebCore/history/BackForwardCache.cpp

```diff
@@ -576,4 +577,19 @@ void BackForwardCache::prune(PruningReason pruningReason)
     }
 }
 
+void BackForwardCache::clearEntriesForOrigins(const HashSet<RefPtr<SecurityOrigin>>& origins)
+{
+    for (auto it = m_items.begin(); it != m_items.end();) {
```

Comment:
Does a modern C++ loop not work for m_items?

Reply:
> I think this looks good, but I worry we may modify the container while iterating, and get weird bugs.
> 
> Also: Could this test be contributed to WPT?

The spec does not mandate the clear this particular cache at the moment. I just chose to because I think it makes sense here.